speed-transition.lua

```diff
@@ -62,35 +97,62 @@ function check_position(_, position)
 end
 
 function skipval()
-   local skipval = mp.get_property_native("demuxer-cache-duration", 5)
-   skipval = clamp(skipval, 1, 5)
+   local skipval = mp.get_property_native("demuxer-cache-duration", 0)
    if nextsub > 0 then
-      if nextsub-skipval-leadin <= 0 or directskip then
-         skipval = nextsub - leadin
+      if directskip then
+         skipval =  nextsub - leadin
+      elseif nextsub - skipval - leadin <= 0 then
+         skipval =  clamp(nextsub - leadin, 0, maxskip)
+      else
+         skipval =  clamp(skipval, 0, maxskip)
       end
+   elseif directskip then
+      skipval = clamp(skipval - leadin, 1, nil)
    else
-      skipval = clamp(skipval-leadin, 0, nil)
-      if skipval == 0 then
-         skipval = clamp(skipval-1, 1, nil)
-      end
+      skipval = clamp(skipval - leadin, 1, maxskip)
    end
    return skipval
 end
 
+firstskip = true   --make the first skip in skip mode not have to wait for skipdelay
+delayEnd = true
 function speed_transition(_, sub)
+   if sub~=nil and shouldIgnore(sub) then
+      sub = ""
+   end;
    if state == 0 then
       if sub == "" then
          last_speedup_zone_begin = speedup_zone_begin
          nextsub, shouldspeedup, speedup_zone_begin = check_should_speedup()
          mark = speedup_zone_begin
-         speedup_zone_end = mark+nextsub
-         if shouldspeedup then
+         speedup_zone_end = mark + nextsub
+         if shouldspeedup or (skipmode and not firstskip) then
             local temp_disable_skipmode = false
             if last_speedup_zone_begin and mark < last_speedup_zone_begin then
                temp_disable_skipmode = true
             end
             if skipmode and not temp_disable_skipmode and mp.get_property("pause") == "no" then
-               mp.commandv("no-osd", "seek", skipval(), "relative", "exact")
+               if firstskip or skipdelay == 0 then
+                  mp.commandv("no-osd", "seek", skipval(), "relative", "exact")
+                  firstskip = false
+               elseif delayEnd then
+                  delayEnd = false
+                  mp.add_timeout(skipdelay, function()
+                     delayEnd = true
+                     if mp.get_property("pause") == "no" then
+                        nextsub, shouldskip = check_should_speedup()
+                        if shouldskip or nextsub > leadin then
+                           local tSkip = skipval()
+                           currentSub = mp.get_property("sub-text")
+                           if tSkip > 1 / mp.get_property("estimated-vf-fps") and (currentSub == "" or shouldIgnore(currentSub)) then
+                              mp.commandv("no-osd", "seek", tSkip, "relative", "exact")
+                           else 
+                              firstskip = true
+                           end
+                        end
+                     end
+                  end)
+               end
```

Comment:
I hoped for a less confusing way to write this but it's effective in making the first skip not have to wait for delay. Also L147 will prevent the skip if it is shorter than the length of a frame

Reply:
Does it even make sense to make short skips, since seek 'exact' is quite slow. Imo limiting tSkip to be above >1 second makes more sense, then even 1 may be too low. What do you say?

Reply:
Ya, I was thinking to limit it by the leadin value again even though our skip already calculates it. Maybe make a configurable minSkip value?

Reply:
Sounds good. What should be the default?

Reply:
default
`minSkip = leadin`
😅